magento/pwa-studio
# V4P3: refactor: Move most Venia components to separate UI library

## Description

Move most Venia UI into a consumable library, `@magento/venia-ui`. Reduce `@magento/venia-concept` footprint down to a "shell" which imports and presents the `App` object from `@magento/venia-ui`.

This decoupling allows for scaffolding (in an upcoming PR) to produce a very small starter project, without copying a huge number of files into the fresh project directory for the developer to maintain.

- Add `venia-ui` workspace.
- Copy most Venia code into `venia-ui` workspace.
- Change the files remaining in `venia-concept` to import their components from `venia-ui`.
- Break `upward.yml` into an un-opinionated piece of boilerplate in `venia-concept`, and all the Venia-specific implementations in `venia-ui/upward.yml`. Use new `UpwardIncludePlugin` to merge UPWARD files and add their dependencies to the build.

## Verification Steps

1. From a freshly cloned repo, run `yarn run stats:venia`.
2. When the tree diagram of dependencies appears, note the amount of code contained in node_modules versus in the local app. It should be much less.

## Related Issues
#946 and #947
Closes #1551.

## Proposed Labels for Change Type/Package




## Checklist:



**I'd like to expedite this if possible, because it's a lot of renames and merges will get hairy if we leave it out too long.**
